File: BOJ/java/11723.java
/*
 * 집합
 * https://www.acmicpc.net/problem/11723
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());
        int bit = 0;
        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");
            String op = input[0];
            int index = -1;
            if (!op.equals("all") && !op.equals("empty")) {
                index = Integer.parseInt(input[1]) - 1;
            }

            switch (op) {
                case "add":
                    bit = setBit(bit, index);
                    break;
                case "remove":
                    bit = clearBit(bit, index);
                    break;
                case "check":
                    sb.append(getBit(bit, index) ? 1 : 0).append("\n");
                    break;
                case "toggle":
                    bit = toggleBit(bit, index);
                    break;
                case "all":
                    bit = -1;
                    break;
                case "empty":
                    bit = 0;
                    break;
            }
        }

        System.out.println(sb);
    }

    static boolean getBit(int num, int i) {
        return (num & (1 << i)) != 0;
    }

    static int setBit(int num, int i) {
        return num | (1 << i);
    }

    static int clearBit(int num, int i) {
        return num & (~(1 << i));
    }

    static int toggleBit(int num, int i) {
        return num ^ (1 << i);
    }
}

File: BOJ/kotlin/11723.kt

import java.lang.IllegalStateException

fun main() {
    val sb = StringBuilder()
    val n = readLine()!!.toInt()
    var bit = 0
    repeat(n) {
        val input = readLine()!!.split(" ")
        val op = input[0]
        val index = if (op !in listOf("all", "empty")) input[1].toInt() - 1 else -1
        bit = when (op) {
            "add" -> setBit(bit, index)
            "remove" -> clearBit(bit, index)
            "toggle" -> toggleBit(bit, index)
            "check" -> {
                sb.append(if (getBit(bit, index)) 1 else 0).append("\n")
                bit
            }
            "all" -> -1
            "empty" -> 0
            else -> throw IllegalStateException()
        }
    }
    println(sb)
}

fun getBit(num: Int, i: Int) = (num and (1 shl i)) != 0
fun setBit(num: Int, i: Int) = num or (1 shl i)
fun clearBit(num: Int, i: Int) = num and (1 shl i).inv()
fun toggleBit(num: Int, i: Int) = num xor (1 shl i)

